uniffi_bindgen/src/interface/types/mod.rs

```diff
@@ -35,6 +35,18 @@ pub(super) use resolver::{resolve_builtin_type, TypeResolver};
 /// Represents all the different high-level types that can be used in a component interface.
 /// At this level we identify user-defined types by name, without knowing any details
 /// of their internal structure apart from what type of thing they are (record, enum, etc).
+//
+// How to implement a new type:
+//
+// - Add a new Type variant
```

Comment:
I believe we eventually want to get out of the business of adding more variants to this Enum in order to support more types, and it may be worth mentioning that here. That said, it's useful to have the current state of the world documented, even if we're still trying to figure out what the new state of the world can look like, so 👍🏻 to add this.

Reply:
Yeah, I agree.  The way I see it, we want to add some sort of wrapper/adapter type.  However, it's hard to add new types right now.  So I'm hoping to improve that situation then implement the wrapper type.  Then we're at a point where it's now easy to add a new type, but we probably don't need to because we can just wrap an existing type.  It's silly in a way, but I still think it's useful because the work to make it easier to add a new type will also make it easier to understand how the current types work.